### DISCUSSION TITLE
SVG to PNG uses wrong colors after upgrade.

### DISCUSSION THREAD

**Discussion by TheAlmightyGuru** [ImageMagick/ImageMagick#6764] [Help]
### ImageMagick version

6.9.12.97

### Operating system, version and so on

Red Hat Enterprise Linux Server release 7.9

### Description

My web host recently upgraded ImageMagick to version 6.9.12.97. Prior to this, I could use ImageMagick through PHP to convert SVG images to PNG images and everything looked fine. On the new version, the colors are converted to shades of magenta, green, and grayscale. Is it possible their upgrade had a problem?

Here is an example:

![Color problem](https://github.com/ImageMagick/ImageMagick/assets/9964540/59052b6f-cb7e-4c78-9d51-f36128c96444)

**Comment by fmw42** [comment]
Post your SVG file so we can test with it.

**Reply by TheAlmightyGuru** [reply]
Thanks fmw42. Here's what I'm testing with.
![test](https://github.com/ImageMagick/ImageMagick/assets/9964540/801c2ebc-2970-43ce-888e-34240b76b23f)

The problem is occurring on all SVG images, including those that I used in the past and they were converted correctly.

**Reply by TheAlmightyGuru** [reply]
From your idea, I've also tried forcing an RGB colorspace, but it doesn't help. Strangely, trying to force a grayscale colorspace doesn't result in a grayscale image, but still the magenta and green. It's as though the server is always applying its own colorspace.

**Comment by snibgo** [comment]
What is your command, and what is the text output when you insert `-verbose` as the first option?

**Reply by TheAlmightyGuru** [reply]
Hello snibgo. I'm using PHP to perform the conversion. Here is the relevant code:

$im = new Imagick($uploadedFile);
$im->writeImage($downloadFile);

Is there a way to turn on verbose logging in PHP?

**Reply by TheAlmightyGuru** [reply]
If it helps, gere is the contents of $im->identifyImage():

Array
(
    [mimetype] => image/svg+xml
    [format] => SVG (Scalable Vector Graphics)
    [units] => Undefined
    [colorSpace] => sRGB
    [type] => TrueColorAlpha
    [compression] => Undefined
    [fileSize] => 4073B
    [geometry] => Array
        (
            [width] => 450
            [height] => 450
        )

    [resolution] => Array
        (
            [x] => 96
            [y] => 96
        )

    [signature] => 9e2f9f5356a3dacc732fd3d61d3def60cfd91a56cac5824694d539519d678cb9
)

**Comment by fmw42** [comment]
Sorry about the PDF post. I deleted but the email had already gone out.  I misread the title and and realized it was SVG not PDF, but too late.

**Reply by TheAlmightyGuru** [reply]
No worries!

**Comment by fmw42** [comment]
What SVG renderer are/were you using (IM MSVG, RSVG or Inkscape).  I would suggest that you use Inkscape as it is the best of the 3.

**Reply by TheAlmightyGuru** [reply]
I'm kind of stuck using ImageMagick because I have other software written for it, and my web host doesn't allow programs external to PHP called though exec.

**Comment by TheAlmightyGuru** [comment]
I just downloaded "ImageMagick-6.9.12-97-Q16-HDRI-x64-dll.exe" for Windows from the legacy web site (https://legacy.imagemagick.org/script/download.php).

I installed it, opened IMDisplay and loaded the SVG image, and it's giving the same results. This appears to be a problem with the build itself. (Sorry for Windows 7, I'm testing in a virtual machine.)

![IMDisplay](https://github.com/ImageMagick/ImageMagick/assets/9964540/3570cd98-d874-4925-b222-b15099f31243)

And it's not just my test image, here's the Flag of Michigan I pulled off Wikipedia:
![IMDisplay 2](https://github.com/ImageMagick/ImageMagick/assets/9964540/c33741d5-1384-4279-93ce-fc5a2884e2c0)

**Comment by TheAlmightyGuru** [comment]
I was able to convince my web host to upgrade to ImageMagick 7.1.1-19 and the problem went away.

This appears to be a bug in version 6.9.12-97. So, if anyone comes here seeking help for this same problem, upgrade to version 7.

**Comment by thebaptiste** [comment]
I have exactly the same problem converting svg to pdf on linux with release 6.9.13-98. It's still here with last week's release 6.9.13-9... It was ok with 6.9.12-93.
May this should be fixed in one of the subsequent releases of the ImageMagick6 serie, not only in ImageMagick7 ?

**Reply by fmw42** [reply]
It likely depends upon what SVG renderer you are using in IM 7 and in IM 6. I suspect you have two different SVG renderers.  Please check if you have Inkscape installed in IM 7 where it works and in IM 6 you do not and are using the internal MSVG renderer. You can see where it is that or RSVG delegate (if not Inkscape) using convert or magick -list type. Look at the end of the line for SVG.

**Reply by andy96453** [reply]
Have same problem using 6.9.13-14 RSVG 2.42.7
Have reverted back to 6.9.12-93 RSVG 2.42.7 and all OK

**Reply by fmw42** [reply]
Install Inkscape and IM will use it automatically. No need to re-install IM.  It should work better than RSVG.

**Reply by dlemstra** [reply]
I pushed a patch yesterday to fix this inside ImageMagick 6.

**Reply by scottbuscemi** [reply]
@dlemstra Any chance the latest rpm could be uploaded? Our team is blocked from releasing this :)  https://github.com/ImageMagick/ImageMagick6/issues/353

**Reply by dlemstra** [reply]
This was fixed in `6.9.13-15` so the rpm that is now available also includes that fix.

**Reply by scottbuscemi** [reply]
@dlemstra Ah. Even more strange, thanks. Might be a separate color shifting issue then. Will investigate 👍🏼

**Comment by thebaptiste** [comment]
Thanks
Will you release a 6.9.13-15 with this fix ?
